docs/intro.md

```diff
@@ -0,0 +1,1014 @@
+
+When registered to process streaming data, the function is executed once for each new input record as a default. Whereas batch functions are executed exactly once, a registered function's execution may be triggered any number of times in response to the events that drive it.
+
+To try this, we'll return to the maximum computing example and have it executed in response to new data with the [`register()`](functions.md#register) action:
+
+```python
+def maximum(a, x):
+  ''' Returns the maximum '''
+  a = a if a else 0  # initialize the accumulator
+  if x > a:          # if a new maximum was found
+    a = x            # set accumulator to maximum
+  return a           # finally output accumulator
+
+# This function registers to keyspace events
+GB() \
+  .map(lambda x: int(x['value']['age'])) \
+  .accumulate(maximum) \
+  .register('person:*')
+
+## Expected result: ['OK']
+```
+
+By ending a function with the [`register()`](functions.md#register) action and sending it to RedisGears, the engine registers it and will execute it in response to the reader's events. In the case of the **KeysReader**, events are generated every time keys that match the pattern `person:*` are written to the database.
+
+Just `register()`ing the function doesn't trigger its execution, and the "OK" returned in the response serves only to indicate that its registration was successful. Changes to data will trigger execution, which can be done for example with the following Redis command:
+
+```
+HSET person:3 name "Summer Smith" age 17
+```
+
+A registered function is by definition nonblocking, so any results it returns can only be obtained with the `RG.GETRESULTS` command and by specifying its respective execution ID. At this point we know there is just one registered function and that it had executed just once, so we can use `RG.DUMPEXECUTIONS` output to get the last execution's ID and then read the results:
+
+!!! example "Example: dumping executions"
+    ```
+    127.0.0.1:6379> HSET person:3 name "Summer Smith" age 17
+    (integer) 2
+    127.0.0.1:6379> RG.DUMPEXECUTIONS
+    1) 1) "executionId"
+       2) "0000000000000000000000000000000000000000-1"
+       3) "status"
+       4) "done"
+    2) 1) "executionId"
+       2) "0000000000000000000000000000000000000000-0"
+       3) "status"
+       4) "done"
+    127.0.0.1:6379> RG.GETRESULTS 0000000000000000000000000000000000000000-1
+    1) 1) (integer) 17
+    2) (empty array)
+    ```
+
+Note that the executions' list now consists of two entries: the first is the most recent one generated by the registered function, and the second entry is from our previous nonblocking execution of the batch function.
```

Comment:
The printed order is not necessarily the creation order

Reply:
Fixed via 079f280